iothub/device/src/IotHubBaseClient.cs

```diff
@@ -190,45 +190,43 @@ public async Task SendEventBatchAsync(IEnumerable<Message> messages, Cancellatio
         }
 
         /// <summary>
-        /// Sets a new delegate for receiving a message from the device or module queue using a cancellation token.
+        /// Sets a callback for receiving a message from the device or module queue using a cancellation token.
         /// This instance must be opened already.
         /// </summary>
         /// <remarks>
-        /// <para>
-        /// If a delegate is already registered it will be replaced with the new delegate.
-        /// If a null delegate is passed, it will disable the callback triggered on receiving messages from the service.
-        /// </para>
+        /// Calling this API more than once will result in the callback set last overwriting any previously set callback.
+        /// A method callback can be unset by setting <paramref name="messageCallback"/> to null.
         /// </remarks>
-        /// <param name="messageHandler">The delegate to be used when a could to device message is received by the client.</param>
+        /// <param name="messageCallback">The callback to be invoked when a cloud-to-device message is received by the client.</param>
         /// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
         /// <exception cref="InvalidOperationException">Thrown if instance is not opened already.</exception>
         /// <exception cref="OperationCanceledException">Thrown when the operation has been canceled.</exception>
-        public async Task SetMessageHandlerAsync(
-            Func<Message, Task<MessageAcknowledgement>> messageHandler,
+        public async Task SetMessageCallbackAsync(
```

Comment:
I feel like "message" isn't descriptive enough, and we probably should say "CloudToDeviceMessage". Thoughts?

Reply:
This is being used for both device and module clients, so we won't be able to use CloudTo"Device"... We could use CloudToClient, but as Tim pointed out in a previous comment, there's room for this being mis-interpreted since our service client is also a "client".

Reply:
SetMessageCallbackAsync is used for C2D and D2C? By the customer?

Reply:
SetMessageCallbackAsync is being used by both device and module clients to receive messages from service

Reply:
Oh, so in the module case, it isn't always CloudToDevice. Gotcha.

SetIncomingMessageHandlerAsync?

Reply:
I like that, yeah